daemon/checkpoint.go

```diff
@@ -82,6 +90,23 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Cannot checkpoint container %s: %s", name, err)
 	}
 
+	checkpointInfo := checkPoint{
+		Name:    config.CheckpointID,
+		Created: time.Now(),
+		Exit:    config.Exit,
+	}
+	f, err := os.Create(filepath.Join(checkpointDir, "config.json"))
+	if err != nil {
+		os.RemoveAll(checkpointDir)
```

Comment:
It just occurred to me, but does this play nice with custom checkpoint dirs? (i.e.; should it remove the directory in that case?)

Reply:
The alternative would be to check if we created the directory, but that would be dangerous (there could be a race-condition where we checked if it existed before, but meanwhile another session is creating a checkpoint (and the checkpoint dir))

Reply:
Hm, no, I don't think we should add an option for this; also in light of future work for checkpoints is being worked on on containerd.

Reply:
For https://github.com/containerd/containerd/blob/06b9cb35161009dcb7123345749fef02f7cea8e0/runtime/container.go#L367
If write config.json fail, it just return the error.
I think there is very very small opportunity to meet this error, so we just return the error without remove the dir?
If yes, I'll remove this line, and modify the checkpoint list func to ignore this type folder because there is no config.json in it(very very small opportunity).

Reply:
Or move create config.json to the position before call daemon.containerd.CreateCheckpoint?

Reply:
@thaJeztah I have moved the code position, and the checkpoint dir is always created by dockerd, not created by the user(https://github.com/moby/moby/blob/502b6e62ac7e0f437a9c78045a5c92b900e50704/daemon/checkpoint.go#L44), so it will not delete user's exited dir.